src/posix/platform/infra_if.cpp

```diff
@@ -710,7 +710,10 @@ otError InfraNetif::DiscoverNat64Prefix(uint32_t aInfraIfIndex)
         {
             freeaddrinfo(hints);
         }
-        free(reqs[0]);
+        if (reqs[0])
+        {
+            free(reqs[0]);
```

Comment:
I think `free()` can be used with `nullptr` input (would do nothing)?

Reply:
Yes, I agree. If free() is used with nullptr, it does nothing. 
So what do you suggest? Should I remove `if (reqs[0])` or keep to follow format same as done for `hints`.

Reply:
Yes. Thanks.

Reply:
Sorry, I didn't get to which option you agreed.
1. Remove `if (reqs[0])`
2. keep `if (reqs[0])`  as it is by following the format same as done for `hints` as below.
```     
if (hints)
{
    freeaddrinfo(hints);
}
```

Reply:
I would say removing `if()` for the case used with `free()` make sense (since it is supposed to work correctly with `nullptr`).

I am not sure the behavior of `freeaddinfo()`.